api/v1beta1/types_template.go

```diff
@@ -86,10 +86,15 @@ type VnetTemplateSpec struct {
 }
 
 // VnetPeeringsTemplateSpec defines a list of peerings of the newly created virtual network with existing virtual networks.
+// +listType=map
+// +listMapKey=remoteVnetName
 type VnetPeeringsTemplateSpec []VnetPeeringClassSpec
 
 // SubnetTemplateSpec specifies a template for a subnet.
 type SubnetTemplateSpec struct {
+	// Name is the name for this subnet.
+	Name string `json:"name"`
```

Comment:
@CecileRobertMichon @sbueringer @ykakarap I assume this is the change you're talking about. Obviously we can't simply update this type spec without a lot more considerations, but wanted to make sure I understand that conversation

Reply:
Yup exactly

Reply:
I think it should be added inside https://github.com/jackfrancis/cluster-api-provider-azure/blob/2ce64d384564971e96c83718cf9a9aee6ba1ce39/api/v1beta1/types_class.go#L74 no? it's nested inline so in practice same result but it's part of the subnet definition

Reply:
Happy to do either. Should we file an issue to enumerate how we can safely implement this breaking change?

Reply:
FWIW ClusterClass is experimental and behind a feature flag and these types are not used anywhere outside ClusterClass so breaking changes are fair game, as documented in experimental policy

Reply:
Ooohhhhh 💪 

Great, so I'll move the `Name` into the `SubnetClassSpec` type itself, and then audit the code and reference templates and make all the changes.

We'll definitely want to call this out in the release.

Reply:
@CecileRobertMichon We currently have the following struct hierarchies;

AzureCluster > NetworkSpec > Subnets > SubnetSpec (contains Name) > SubnetClassSpec

AzureClusterTemplate > NetworkTemplateSpec > SubnetTemplatesSpec > SubnetTemplateSpec (missing Name) > SubnetClassSpec

I think there are two options:

1. Add the `Name` to `SubnetClassSpec`, drop it from `SubnetSpec`
2. Add the `Name` to `SubnetTemplateSpec`

Given that the idea of `SubnetClassSpec` is to contain the shared fields between AzureCluster and AzureClusterTemplate it would be the right place to keep this pattern, we just should't forget to drop `Name` from `SubnetSpec`)